ledger-core/virtual/integration/deactivate_test.go

```diff
@@ -746,3 +746,165 @@ func TestVirtual_CallMethod_After_Deactivation(t *testing.T) {
+
+	defer commonTestUtils.LeakTester(t)
+
+	cases := []struct {
+		name      string
+		isolation contract.MethodIsolation
+	}{
+		{
+			name:      "Tolerable-Dirty",
+			isolation: tolerableFlags(),
+		},
+		{
+			name:      "Intolerable-Validated",
+			isolation: intolerableFlags(),
+		},
+	}
+	for _, testCase := range cases {
+		t.Run(testCase.name, func(t *testing.T) {
+			mc := minimock.NewController(t)
+
+			server, ctx := utils.NewUninitializedServer(nil, t)
+
+			oneExecutionEnded := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+			runnerMock := logicless.NewServiceMock(ctx, mc, nil)
+			server.ReplaceRunner(runnerMock)
+			server.Init(ctx)
+
+			var (
+				class               = gen.UniqueGlobalRef()
+				objectRef           = reference.NewSelf(server.RandomLocalWithPulse())
+				p1                  = server.GetPulse().PulseNumber
+				deactivateIsolation = contract.MethodIsolation{
+					Interference: contract.CallTolerable,
+					State:        contract.CallDirty,
+				}
+				stateRef = server.RandomGlobalWithPulse()
+			)
+
+			server.IncrementPulseAndWaitIdle(ctx)
+			outgoing := server.BuildRandomOutgoingWithPulse()
+			checkOutgoing := server.BuildRandomOutgoingWithPulse()
+
+			{
+				runnerMock.AddExecutionClassify(checkOutgoing.String(), testCase.isolation, nil)
+				if testCase.isolation.State == contract.CallValidated {
+					runnerMock.AddExecutionMock(checkOutgoing.String()).AddStart(
+						func(ctx execution.Context) {
+							require.Equal(t, objectRef, ctx.Request.Callee)
+							require.Equal(t, []byte(origMem), ctx.ObjectDescriptor.Memory())
+						},
+						&execution.Update{
+							Type:   execution.Done,
+							Result: requestresult.New([]byte("check done"), objectRef),
+						},
+					)
+				}
+			}
+
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+			typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+				require.Equal(t, objectRef, res.Callee)
+				require.Equal(t, checkOutgoing, res.CallOutgoing)
+				if testCase.isolation.State == contract.CallValidated {
+					require.Equal(t, []byte("check done"), res.ReturnArguments)
+				} else {
+					contractErr, sysErr := foundation.UnmarshalMethodResult(res.ReturnArguments)
+					require.NoError(t, sysErr)
+					require.Contains(t, contractErr.Error(), "try to call method on deactivated object")
+				}
+				return false
+			})
+			typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+				require.Equal(t, objectRef, report.Object)
+				require.Equal(t, payload.Inactive, report.Status)
+				require.True(t, report.DelegationSpec.IsZero())
+				require.Equal(t, int32(0), report.UnorderedPendingCount)
+				require.Equal(t, int32(0), report.OrderedPendingCount)
+				require.Nil(t, report.ProvidedContent)
+				return false
+			})
+			typedChecker.VDelegatedCallResponse.Set(func(response *payload.VDelegatedCallResponse) bool {
+				// send delegation request finished with deactivate flag
```

Comment:
Don't we need to check VDelegatedCallResponse?

Reply:
add some checks